test-app/tests/integration/components/file-dropzone-test.gts

```diff
@@ -179,6 +180,28 @@ module('Integration | Component | FileDropzone', function (hooks) {
     assert.verifySteps(['dingus.txt']);
   });
 
+  test('allowFolderDrop=true allows dropping a directory and reads out the content', async function (this: LocalTestContext, assert) {
+    const queue = this.queue;
+    const onDrop = (files: UploadFile[]) => {
+      files.forEach((file) => assert.step(file.name));
+    };
+
+    await render(<template>
+        <FileDropzone
+          class="test-dropzone"
+          @queue={{queue}}
+          @allowFolderDrop={{true}}
+          @onDrop={{onDrop}} />
+      </template>);
+
+    await dragAndDropDirectory('.test-dropzone', 'directory-name', [
+      new File([], 'dingus.txt'),
+      new File([], 'dingus.png'),
+    ]);
+
+    assert.verifySteps(['directory-name', 'dingus.txt', 'dingus.png']);
```

Comment:
Not sure that a directory should be included in the files array from `onDrop`. It's not really an `UploadFile` that is able to be uploaded

We might need to add another property to these callbacks which _only_ has directories (maybe as strings of the directory name)